apps/src/code-studio/components/ExportDialog.jsx

```diff
@@ -206,25 +219,95 @@ class ExportDialog extends React.Component {
   };
 
   componentDidUpdate(prevProps) {
-    const {isOpen, projectUpdatedAt} = this.props;
-    const {exportProjectUpdatedAt} = this.state;
+    const {isOpen, md5SavedSources} = this.props;
+    const {md5PublishSavedSources} = this.state;
     if (isOpen && !prevProps.isOpen) {
       recordExport('open');
-      if (projectUpdatedAt !== exportProjectUpdatedAt) {
-        // The project has changed since we last opened the dialog,
-        // reset our export state, so we will need to export again:
-        this.resetExportState();
+      const md5ApkSavedSources = this.getMd5ApkSavedSources();
+      const sourcesChangedSinceExport =
+        md5ApkSavedSources && md5SavedSources !== md5ApkSavedSources;
+      const sourcesChangedThisInstance =
+        md5PublishSavedSources && md5SavedSources !== md5PublishSavedSources;
+      if (sourcesChangedSinceExport) {
+        // Also cancel a build that may have started before this page was reloaded:
+        this.cancelIfPreexistingApkBuild();
```

Comment:
This is outside the scope of this review, but I'm noting it here so we revisit:
It seems like we should ask the user before canceling a build. Especially since it's a 15 minute operation.

Reply:
Fair point, the original design is that when you leave the dialog, you're walking away from the build. If you come back and it is still going, that's more of a pleasant surprise.

I'll discuss with @ryansloan how to deal with the complexity of allowing a student to complete an existing build even though the project has changed since then.

Reply:
👍